frontend/src/pages/projects/screens/detail/storage/StorageTable.tsx

```diff
@@ -15,20 +19,66 @@ type StorageTableProps = {
 const StorageTable: React.FC<StorageTableProps> = ({ pvcs, refresh, onAddPVC }) => {
   const [deleteStorage, setDeleteStorage] = React.useState<PersistentVolumeClaimKind | undefined>();
   const [editPVC, setEditPVC] = React.useState<PersistentVolumeClaimKind | undefined>();
+  const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
+  const [alertStatuses, setAlertStatuses] = React.useState<Record<string, boolean>>({});
+
+  const isDeprecatedAlert = React.useMemo(
+    () => Object.values(alertStatuses).includes(true),
+    [alertStatuses],
+  );
+  const [storageClasses, storageClassesLoaded] = useStorageClasses();
+  const storageClassConfigMap = React.useMemo(
+    () =>
+      storageClasses.reduce((acc: Record<string, StorageClassConfig | undefined>, sc) => {
+        acc[sc.metadata.name] = getStorageClassConfig(sc);
+
+        return acc;
+      }, {}),
+    [storageClasses],
+  );
+  const getStorageColumns = () => {
+    let storageColumns = columns;
+
+    if (!isStorageClassesAvailable) {
+      storageColumns = columns.filter((column) => column.field !== 'storage');
+    }
+    return storageColumns;
+  };
 
   return (
     <>
+      {isDeprecatedAlert && isStorageClassesAvailable && (
+        <Alert
+          data-testid="storage-class-deprecated-alert"
+          variant="warning"
+          isInline
+          title="Deprecated storage class"
+          actionClose={
+            <AlertActionCloseButton
+              data-testid="storage-class-deprecated-alert-close-button"
+              onClose={() => setAlertStatuses({})}
```

Comment:
you will now have to have a state for is the alert has been dismissed. dont tie it into the use memo